proto/rill/runtime/v1/resources.proto

```diff
@@ -386,6 +388,40 @@ message ThemeSpec {
 
 message ThemeState {}
 
+message Chart {
+  ChartSpec spec = 1;
+  ChartState state = 2;
+}
+
+message ChartSpec {
+  string title = 1;
+  string query_name = 2;
+  string query_args_json = 3;
+  string vega_lite_config = 4;
+}
+
+message ChartState {}
+
+message Dashboard {
+  DashboardSpec spec = 1;
+  DashboardState state = 2;
+}
+
+message DashboardSpec {
+  string title = 1;
+  int64 grid_columns = 2;
+  int64 grid_rows = 3;
+  repeated DashboardComponent components = 4;
+}
+
+message DashboardState {}
+
+message DashboardComponent {
+  string chart = 1;
+  optional int64 columns = 2;
+  optional int64 rows = 3;
```

Comment:
Hmm not sure if this will work if we want to differentiate between `(0,0)` and not specified. Honestly lets keep this as is for now. Once we have a prototype from UI side we can tweak this.

Reply:
But lets keep this not optional for now. That makes the most sense and we can change to optional if needed.